spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkAction.java

```diff
@@ -85,6 +88,7 @@
   private static final Logger LOG = LoggerFactory.getLogger(BaseSparkAction.class);
   private static final AtomicInteger JOB_COUNTER = new AtomicInteger();
   private static final int DELETE_NUM_RETRIES = 3;
+  private static final int DELETE_GROUP_SIZE = 100000;
```

Comment:
Honestly I don't see a lot of use cases for changing this. It's really just to set a limit on how many records we pull into memory at a time.

Reply:
yeah that makes sense, I was thinking that since the deletion is done on the driver people would want control over how much is brought into memory, but we don't need to overcomplicate it. 100000 seems like a good default

Reply:
Yeah if at all possible I want to just remove configuration options and instead just have things that always work

Reply:
Yeah that makes sense, in that case we can just leave it as is.

Reply:
I don't think we would want to make it configurable but can 100K be too many? This will be used in the path that collects chunk by chunk to the driver to avoid OOM, which is supposed to help small drivers or scenarios with lots of files to delete. Do we think the records are tiny so 100K won't cause any issues?

I think S3FileIO would send 250 files in a single bulk delete by default.

Reply:
This should just be paths, so 100K 1Kb Strings would still only be 100MB.

Reply:
We would need to make sure this logic works with the streaming iterator in Spark.